src/App.js

```diff
@@ -14,11 +17,137 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    listIsVisible: true,
+    buttonIsVisible: false,
+    products: goodsFromServer,
+    initialState: goodsFromServer,
+    selectedElementLength: 1,
+  }
+
+  showList = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      listIsVisible: !prevState.listIsVisible,
+      buttonIsVisible: !prevState.buttonIsVisible,
+    }));
+  }
+
+  reverse = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      products: [...prevState.products].reverse(),
+    }));
+  }
+
+  sortAlphabet = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      products: [...prevState.products].sort(),
+    }));
+  }
+
+  reset = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      products: prevState.initialState,
+    }));
+  }
+
+  sortByLength = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      products: [...prevState.products]
+        .sort((prod1, prod2) => prod1.length - prod2.length),
+    }));
+  }
+
+  sortBySelect = ({ target }) => {
+    this.setState(prevState => ({
+      products: prevState.initialState
+        .filter(product => product.length >= target.value),
+    }));
+  }
+
+  render() {
+    const { listIsVisible, products, buttonIsVisible } = this.state;
+    const productsCopy = [...products];
+    const selectArray = new Array(10).fill(0);
```

Comment:
create array outside of the component, no need to create It every time when your component is re-rendering